### ISSUE TITLE
"Internal server error" when adding explicit nameservers

### ISSUE THREAD

**Issue by bortzmeyer** [zonemaster/zonemaster#925]
If I add explicit authoritative name servers via the Web interface (without IP addresses) and click Check, I get:

Internal server error

Tested on both zonemaster.fr and zonemaster.net.

**Comment by matsduf** [comment]
@bortzmeyer, thank you. I will investigate.

**Comment by pnax** [comment]
I can replicate.

1. go to zonemaster.net or zonemaster.fr
2. enter `net` as a domain
3. in the options add `ns.nic.se` in the NS field
4. leave the IP field empty
5. open the console
6. run the check
7. see the «Internal server error» message
8. look at the error in the console
```
"/nameservers/0/ip: String does not match ^[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}$|^([0-9A-Fa-f]{1,4}:[0-9A-Fa-f:]{1,}(:[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3})?)$|([0-9A-Fa-f]{1,4}::[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3})$.\n"
```

Here is the request sent to the API :
```json
{
  "jsonrpc": "2.0",
  "id": 1612281193901,
  "method": "start_domain_test",
  "params": {
    "ipv4": true,
    "ipv6": true,
    "profile": "default",
    "domain": "net",
    "ds_info": [],
    "nameservers": [
      {
        "ns": "ns.nic.se",
        "ip": ""
      }
    ],
    "client_version": "3.2.0",
    "client_id": "Zonemaster GUI"
  }
}
```

Looking at the `nameserver` entry, both `ns` and `ip` are defined. However `ip` holds an empty string. Hence the error because the Backend will see the `ip` field and [look for an IP address in this field.

Removing the `ip` field and manually sending the command via curl works.

I think the GUI should not send the `ip` field if it is empty (the same applies to the `ns` field).

Additionally, it would be nice to be able to display better error messages to the user.

**Comment by matsduf** [comment]
And I think the backend should ignore empty `ip` field. I think this is a backend issue.

**Comment by matsduf** [comment]
A work-around has been implemented by  zonemaster/zonemaster-backend#702 and will be included in the v2020.2 release.
